advanced/detect-maestro-in-your-app.md

```diff
@@ -16,4 +16,20 @@ fun isMaestro(): Boolean {
 }
 ```
 
-An alternative is to use [arguments](../api-reference/commands/launchapp.md#launch-arguments) and have your app detect a particular parameter to indicate Maestro's usage, e.g. `isE2ETest`.
+Depending on various variables in your testing setup, the code block above might give you a `NetworkOnMainThreadException `. Get rid of the exception by moving the boolean check to a Coroutine background thread for the `Socket` usage to work without a crash:
```

Comment:
This is quite wordy. How about something like this?

### NetworkOnMainThreadException
On Android, you may encounter a NetworkOnMainThreadException. This means you'll need to move the call above to a background thread. As an example:

```kotlin
CoroutineScope(Dispatchers.IO).launch {
    if (isMaestro()) {
        // do maestro things
    }
}
```

Reply:
Ok, I updated the pr based on your thoughts.

I do think that _not_ mentioning returning to the UI thread runs the risk of confusing Android devs who don't know about IO/UI threads. They might think it's ok to run everything within the IO launch {} wrapper. They might not know about `withContext(Main)`/`runOnUiThread{}`. Not sure how likely that would be, but food for thought. You can update in a new pr if you think it's worth adding.

Reply:
Feel free to merge this whenever you want. Doesn't look like I can do it.

Reply:
I do think it's reasonable to expect Android developers to be aware of this but we'll keep an eye out for any confusion in the community regarding this.